Lib9c/Model/Item/Inventory.cs

```diff
@@ -442,7 +442,33 @@ public static bool IsMaterialRemovable(Item item, int id, long blockIndex)
         }
 
         /// <summary>
-        /// Remove a material from the inventory.
+        /// Checks if the given item is non-tradable material and can be removed from the inventory.
+        /// </summary>
+        /// <param name="item">The item to check for removal.</param>
+        /// <param name="id">The ID of the material to remove.</param>
+        /// <returns>Returns a boolean indicating whether the item is removable.</returns>
+        public static bool IsMaterialRemovable(Item item, int id)
+        {
+            if (item.Locked)
+            {
+                return false;
+            }
+
+            if (item.item is Material material && material.Id == id)
+            {
+                if (material is TradableMaterial)
+                {
+                    return false;
+                }
+
+                return true;
+            }
+
+            return false;
+        }
```

Comment:
It seems not to access any other methods, properties of `Inventory`. Shouldn't it be a static method and extracted from `Inventory` class?

Reply:
Since it is already defined as a static method, relocating it should not cause any significant issues.